omod/src/main/java/org/openmrs/module/adminui/page/controller/account/ChangeSecretQuestionPageController.java

```diff
@@ -0,0 +1,163 @@
+/**
+ * The contents of this file are subject to the OpenMRS Public License
+ * Version 1.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ * http://license.openmrs.org
+ *
+ * Software distributed under the License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
+ * License for the specific language governing rights and limitations
+ * under the License.
+ *
+ * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
+ */
+
+package org.openmrs.module.adminui.page.controller.account;
+
+import org.openmrs.api.UserService;
+import org.openmrs.api.context.Context;
+import org.openmrs.messagesource.MessageSourceService;
+import org.openmrs.module.adminui.AdminUiConstants;
+import org.openmrs.ui.framework.annotation.BindParams;
+import org.openmrs.ui.framework.annotation.MethodParam;
+import org.openmrs.ui.framework.annotation.SpringBean;
+import org.openmrs.ui.framework.page.PageModel;
+import org.springframework.context.MessageSource;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.ObjectError;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.List;
+
+public class ChangeSecretQuestionPageController {
```

Comment:
I meant that some of the comments in the controller before apply to this too e.g not needed the MessageSource parameter,  and moving the line that adds the errors object to model to the sendErrorMessage method